packages/ui/src/working-groups/hooks/useWithdrawMembershipTransaction.ts

```diff
@@ -0,0 +1,15 @@
+import { useMemo } from 'react'
+
+import { useApi } from '../../common/hooks/useApi'
+import { groupExtrinsics } from '../types'
+
+export function useWithdrawMembershipTransaction(groupName: string) {
+  const { api } = useApi()
+
+  return useMemo(() => {
+    if (api && groupName in groupExtrinsics) {
+      const name = groupName as keyof typeof groupExtrinsics
+      return api.tx[groupExtrinsics[name]].withdrawApplication
+    }
+  }, [api, groupName])
+}
```

Comment:
I'd say that this is too narrow for a hook -> There's ~10 extrinsic to handle so that would mean 10 atomic hooks.

I think that we might be better of with a pure function that will return proper extrinsic:

```typescript
const tx = getGroup(api, 'forum').withdrawAppliction()
```

If you manage to have proper type on the `tx` that would be an awesome bouns (I didn't check if that is not a case already ;) )